plugins/dualGovernance/hooks/useCanCreateProposal.tsx

```diff
@@ -56,29 +80,14 @@ export function useCanCreateProposal() {
     contractReads?.[2]?.status,
   ]);
 
+  useEffect(() => {
+    setHasCreatePermission(!!hasCreatePermissionData);
+  }, [hasCreatePermissionStatus]);
```

Comment:
Question here as well. Why do we set the query response into state instead of relying on the raw response data + response status?

Reply:
I understood that we want to ensure we return a boolean instead of `[boolean | unknown]` (rn is just `unknown` due to the ABI problem). Probably is better to change the return type to one that is more "hook alike" like the one you mentioned. But I'll recommend leaving that change for another ticket/PR specific for that since it might be happening in more places.

Reply:
Oh, I see, makes sense